experimental/local_file_system_memory/local_filesystem_memory.py

```diff
@@ -0,0 +1,260 @@
+        llm = ChatOpenAI(**kwargs)
+        llm_predictor = LLMPredictor(llm=llm)
+        return index.query(prompt, llm_predictor=llm_predictor, similarity_top_k=top_k)
+
+    def __get_docs_from_strings(self, strings):
+        """Get a list of Document objects from a list of strings."""
+        if isinstance(strings, str):
+            docs = [Document(strings)]
+        elif isinstance(strings, list):
+            if isinstance(strings[0], str):
+                docs = [Document(doc) for doc in strings]
+            else:
+                docs = strings
+        return docs
+    
+    def __get_summaries_from_docs(self, docs):
+        """Get a list of summaries from a list of Document objects."""
+        summaries = []
+        for doc in docs:
+            index = GPTSimpleVectorIndex([doc])
+            resp = self._query_index(index, "Summarize:")
+            summary = resp.response
+            summaries.append(summary)
+        return summaries
+
+    def __get_summary_description(self, summary_docs) -> str:
+        """Get a summary description from a list of Documentt objects."""
+        logger.info("Generating description for index...")
+        summary_index = GPTListIndex(summary_docs)
+        resp = summary_index.query("Summarize the documents.")
+        description = resp.response
+        return description
+
+    """CRUD and related public methods for indexes."""
+
+    @property
+    def metadata(self):
+        """Get the metadata."""
+        with open(self.metadata_file, "r") as f:
+            return json.load(f)
+
+    def add_index(self, name, index, description=None):
+        """Add an index to memory."""
+        logger.info("Adding index to memory...")
+        key = list(index._docstore.docs.keys())[0]
+        docs_dict = index._docstore.docs[key].nodes_dict
+        docs = []
+        for key in docs_dict.keys():
+            docs.append(Document(docs_dict[key].text))
+        
+        summaries = self.__get_summaries_from_docs(docs)
+        summary_docs = [Document(summary) for summary in summaries]
+        if description is None:
+            description = self.__get_summary_description(summary_docs)
+
+        metadata = self.metadata
+        # Ensure that the index name is unique.
+        for index_description in metadata["index_descriptions"]:
+            if index_description["name"] == name:
+                logger.info("Index name already exists.")
+                return
+
+        # Update the metadata.
+        metadata["index_count"] += 1
+        metadata["index_descriptions"].append(
+            {
+                "name": name,
+                "description": description,
+            }
+        )
+        self.__save_metadata(metadata)
+
+        # Save the index to disk and add it to the indexes dictionary.
+        index.save_to_disk(self.__get_index_path(name))
+        self.indexes[name] = index
+
+    def create_index(self, name, docs, description=None):
```

Comment:
would the user be able to specify kwargs for configuring the index (e.g. service_context, etc.)? could be good to have a create_kwargs field that gets passed to GPTSimpleVectorIndex during construction